lib/workload/stateful/database/component.ts

```diff
@@ -0,0 +1,47 @@
+import { Construct } from 'constructs';
+import { aws_ec2, aws_rds, RemovalPolicy } from 'aws-cdk-lib';
+import { AuroraPostgresEngineVersion, Credentials } from 'aws-cdk-lib/aws-rds';
+import { IVpc } from 'aws-cdk-lib/aws-ec2';
+
+export interface DatabaseProps {
+  clusterIdentifier: string;
+  defaultDatabaseName: string;
+  version: AuroraPostgresEngineVersion;
+  parameterGroupName: string;
+  username: string;
+}
+
+export class DatabaseConstruct extends Construct {
+  constructor(scope: Construct, id: string, vpc: IVpc, props: DatabaseProps) {
+    super(scope, id);
+
+    const secret = new aws_rds.DatabaseSecret(this, id + 'Secret', {
+      // username: 'admin',
+      username: props.username,
+    });
+
+    new aws_rds.ServerlessCluster(this, id + 'Cluster', {
+      vpc: vpc,
+      vpcSubnets: {
+        subnetType: aws_ec2.SubnetType.PRIVATE_ISOLATED,
+      },
+      engine: aws_rds.DatabaseClusterEngine.auroraPostgres({
+        version: props.version,
+      }),
+      parameterGroup: aws_rds.ParameterGroup.fromParameterGroupName(
+        this,
+        id + 'ParameterGroup',
+        props.parameterGroupName
+      ),
+      removalPolicy: RemovalPolicy.DESTROY,
+      scaling: {
+        // autoPause: Duration.seconds(60),
+        minCapacity: aws_rds.AuroraCapacityUnit.ACU_1,
```

Comment:
Is this for serverless v1 or v2? I think v2 is not fully supported in CDK yet.

You could potentially use serverless v2 by manually specifying the `aws_rds.CfnDBCluster` engineMode and creating a writer instance, similar to [here](https://github.com/umccr/elsa-data-aws-infrastructure/blob/dc19291241e73618c350665b0b0340fb55ff987d/rds/serverless-base-database.ts#L60-L71) and [here](https://github.com/umccr/elsa-data-aws-infrastructure/blob/dc19291241e73618c350665b0b0340fb55ff987d/rds/serverless-base-database.ts#L88-L94). That way, you could set the minCapacity to 0.5, which might be better? Although, it would scale slower to a higher ACU then.

Reply:
Thanks for pointers, Marko. Absolutely we gotta do that. Here is, just a placeholder code for now, yes.